core/src/banking_stage/transaction_scheduler/receive_and_buffer.rs

```diff
@@ -405,8 +407,57 @@ impl TransactionViewReceiveAndBuffer {
 
         let mut num_received = 0usize;
         let mut num_buffered = 0usize;
+        let mut num_dropped_on_status_age_checks = 0usize;
         let mut num_dropped_on_capacity = 0usize;
         let mut num_dropped_on_receive = 0usize;
+
+        // Create temporary batches of transactions to be age-checked.
+        let mut transaction_ids = ArrayVec::<_, EXTRA_CAPACITY>::new();
+        let lock_results: [_; EXTRA_CAPACITY] = core::array::from_fn(|_| Ok(()));
+        let mut error_counters = TransactionErrorMetrics::default();
+
+        let mut run_status_age_checks =
+            |container: &mut TransactionViewStateContainer,
+             transaction_ids: &mut ArrayVec<usize, 64>| {
+                // Temporary scope so that transaction references are immediately
```

Comment:
complexity here can go away once we have `Bytes` backed transactions coming from upstream, since we do not need to do the weird "insert to map only" pattern.